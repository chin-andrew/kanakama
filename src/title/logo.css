.logo-container {
  border: 4px solid black;
  margin: 16px;
  width: 150px;
}

.logo-image {
  height: 75px;
  width: 75px;
}