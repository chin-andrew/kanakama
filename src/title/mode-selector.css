.mode-selector {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: 20px;
}

.mode-text {
  margin-bottom: 8px;
}

.mode-button {
  border: 2px solid black;
  background: none;
  cursor: pointer;
  font-size: 16px;
  letter-spacing: 0.3px;
  margin: 4px;
  padding: 12px 40px;
  width: 200px;
}

.mode-button:hover {
  border: 2px solid black;
  background-color: darkgrey;
}