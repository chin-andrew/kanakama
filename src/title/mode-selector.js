import React from 'react';

export default (onClick) => {
  return (
    <div>
      <div>What would you like to practice?</div>
      <button onClick={() => onClick('hiragana')}>Hiragana</button>
      <button onClick={() => onClick('katakana')}>Katakana</button>
      <button onClick={() => onClick('all')}>Hiragana and Katakana</button>
    </div>
  )
}