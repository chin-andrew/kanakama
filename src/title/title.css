.title-container {
  display: flex;
  flex-direction: column;
  align-items: center
}

.title {
  font-size: 24px;
  margin: 16px;
}