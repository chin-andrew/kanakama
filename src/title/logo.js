import React from 'react';
import fetchKanaImage from '../question/image';
import './logo.css';

export default () => {
  return (
    <div className="logo-container">
      {fetchKanaImage('./hiragana-ka.svg', 'logo-image')}
      {fetchKanaImage('./hiragana-na.svg', 'logo-image')}
      {fetchKanaImage('./hiragana-ka.svg', 'logo-image')}
      {fetchKanaImage('./hiragana-ma.svg', 'logo-image')}
    </div>
  )
}