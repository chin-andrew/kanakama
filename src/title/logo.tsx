import React from 'react';
import styled from 'styled-components';
import fetchKanaImage from '../components/image';
import './logo.css';

export const LogoContainer = styled.div`
  border: 4px solid black;
  margin: 16px;
  width: 150px;
`;

export default () => {
  return (
    <LogoContainer>
      {fetchKanaImage('./hiragana-ka.svg', 'logo-image')}
      {fetchKanaImage('./hiragana-na.svg', 'logo-image')}
      {fetchKanaImage('./hiragana-ka.svg', 'logo-image')}
      {fetchKanaImage('./hiragana-ma.svg', 'logo-image')}
    </LogoContainer>
  );
};
