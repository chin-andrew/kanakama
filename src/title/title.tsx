import React from 'react';
import modeSelector from './mode-selector';
import logo from './logo';
import './title.css';

export default (onClick: Function) => {
  return (
    <div className="title-container">
      {logo()}
      <div className='title'> ｋａｎａｋａｍａ </div>
      {modeSelector(onClick)}
    </div>
  )
}