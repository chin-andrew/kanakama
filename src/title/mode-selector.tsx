import React from 'react';

import { EModes } from '../types/mode';
import './mode-selector.css';

export default (onClick: Function) => {
  return (
    <div className="mode-selector">
      <div className='mode-text'>What would you like to practice?</div>
      <button className="mode-button" onClick={() => onClick(EModes.hiragana)}>Hiragana</button>
      <button className="mode-button" onClick={() => onClick(EModes.katakana)}>Katakana</button>
      <button className="mode-button" onClick={() => onClick(EModes.all)}>Hiragana and Katakana</button>
    </div>
  )
}