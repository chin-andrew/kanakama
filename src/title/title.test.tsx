import renderer from 'react-test-renderer';
import sinon from 'sinon';
import title from './title';
import * as logo from './logo';

describe('title tests', () => {

  let stub: any;
  beforeEach(() => {
    stub = sinon.stub(logo, 'default'); // Need to stub logo call due to require.context
  });

  afterEach(() => {
    stub.restore();
  });

  it('renders a title page', () => {
    const tree = renderer.create(title(() => Object)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
