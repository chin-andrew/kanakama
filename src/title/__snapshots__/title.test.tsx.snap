// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`title tests renders a title page 1`] = `
<div
  className="sc-ifAKCX bNJehV"
  id="title-container"
>
  <span
    className="sc-EHOje ckwPzS"
  >
     ｋａｎａｋａｍａ 
  </span>
  <div
    className="sc-bwzfXH dVfpzR"
  >
    <span
      className="sc-htpNat joCuXP"
    >
      What would you like to practice?
    </span>
    <button
      className="sc-bdVaJa eXCJkO"
      onClick={[Function]}
    >
      Hiragana
    </button>
    <button
      className="sc-bdVaJa eXCJkO"
      onClick={[Function]}
    >
      Katakana
    </button>
    <button
      className="sc-bdVaJa eXCJkO"
      onClick={[Function]}
    >
      Hiragana and Katakana
    </button>
  </div>
</div>
`;
